src/features/board/components/post/PostChecklistContent.tsx

```
@@ -36,11 +41,27 @@ function isAllAnswered(state: ChecklistState): boolean {
+  const { mutate: createPost, isPending: isCreatePending } = useCreatePost();
+  const { mutate: updatePost, isPending: isUpdatePending } = useUpdatePost(postId ?? 0);
+  const { mutate: updateChecklist, isPending: isChecklistPending } = useUpdateUserChecklist();
 
-  // TODO: 실제 API 연동 시 온보딩에서 저장한 사용자 데이터를 초기값으로 로드
   const [answers, setAnswers] = useState<ChecklistState>(getInitialState);
+  const [originalAnswers, setOriginalAnswers] = useState<ChecklistState | null>(null);
+
+  useEffect(() => {
+    if (checklistData && originalAnswers === null) {
+      const loaded = apiChecklistToFormState(checklistData);
+      setAnswers(loaded);
```

ESLint

[error] Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

/home/jailuser/git/src/features/board/components/post/PostChecklistContent.tsx:58:7
  56 |     if (checklistData && originalAnswers === null) {
  57 |       const loaded = apiChecklistToFormState(checklistData);
> 58 |       setAnswers(loaded);
     |       ^^^^^^^^^^ Avoid calling setState() directly within an effect
  59 |       setOriginalAnswers(loaded);
  60 |     }
  61 |   }, [checklistData, originalAnswers]);

(react-hooks/set-state-in-effect)